src/store/user/thunk.js

```diff
@@ -0,0 +1,28 @@
+import { getCurrentUser, logIn, logOut } from '../../services';
+import { logUserIn, logUserOut, setCurrentUserName } from './actionCreators';
+
+export const thunk_logUserIn = (user, history) => async (dispatch) => {
+	await logIn(user)
+		.then((response) => {
+			const token = response.data.result;
+			const name = response.data.user.name;
+			localStorage.setItem('userName', name);
+			localStorage.setItem('token', token);
+			dispatch(logUserIn({ ...user, name: name, token: token }));
+		})
+		.then(() => history.push('/courses'));
+};
+
+export const thunk_logUserOut = (token, history) => async (dispatch) => {
+	await logOut(token).then(() => {
+		localStorage.clear();
+		dispatch(logUserOut());
+		history.push('/login');
+	});
+};
+
+export const thunk_getCurrentUserName = (token) => async (dispatch) => {
```

Comment:
better to make it getUser and set not only name but also a role because after page refresh I don't see controls when I logged as admin

Reply:
Fixed!